tests/tst_onedriveimage/main.cpp

```diff
@@ -0,0 +1,538 @@
+#include <QtSql/QSqlDatabase>
+#include <QtSql/QSqlQuery>
+
+class OneDriveImageTest: public QObject
+{
+    Q_OBJECT
+private:
+
+private slots:
+    // Perform some cleanups
+    // we basically remove the whole ~/.local/share/system/privileged. While it is
+    // damaging on a device, on a desktop system it should not be much
+    // damaging.
+    void initTestCase()
+    {
+        QStandardPaths::enableTestMode(true);
+
+        QDir dir (PRIVILEGED_DATA_DIR);
+        dir.removeRecursively();
+    }
+
+    void testAddUser()
+    {
+        OneDriveImagesDatabase database;
+
+        QDateTime time1 (QDate(2013, 1, 2), QTime(12, 34, 56));
+
+        database.addUser("a", time1, "c");
+        database.commit();
+        database.wait();
+        QCOMPARE(database.writeStatus(), AbstractSocialCacheDatabase::Finished);
+
+        QString dataType = SocialSyncInterface::dataType(SocialSyncInterface::Images);
+        QString dbFile = QLatin1String("onedrive.db"); // DB_NAME in onedriveimagesdatabase.cpp
+
+
+        QSqlDatabase checkDb = QSqlDatabase::addDatabase("QSQLITE");
+        checkDb.setDatabaseName(QString("%1/%2/%3").arg(PRIVILEGED_DATA_DIR, dataType, dbFile));
+        QVERIFY(checkDb.open());
+
+        // Check if the user has been written
+        QSqlQuery query (checkDb);
+        query.prepare("SELECT userId, updatedTime, userName "\
+                      "FROM users");
+        QVERIFY(query.exec());
+        QVERIFY(query.next());
+        QCOMPARE(query.value(0).toString(), QLatin1String("a"));
+        QCOMPARE(query.value(1).toUInt(), time1.toTime_t());
+        QCOMPARE(query.value(2).toString(), QLatin1String("c"));
+        QVERIFY(!query.next());
+
+        // Check if we can retrieve the user
+        OneDriveUser::ConstPtr user = database.user(QLatin1String("a"));
+        QCOMPARE(user->userId(), QLatin1String("a"));
+        QCOMPARE(user->updatedTime(), time1);
+        QCOMPARE(user->userName(), QLatin1String("c"));
+        QCOMPARE(user->count(), -1);
+
+        QDateTime time2 (QDate(2012, 3, 4), QTime(10, 11, 12));
+
+        // Let's write another user
+        database.addUser("d", time2, "f");
+        database.commit();
+        database.wait();
+        QCOMPARE(database.writeStatus(), AbstractSocialCacheDatabase::Finished);
+
+        // Check if we can retrieve everybody
+        database.queryUsers();
+        database.wait();
+        QList<OneDriveUser::ConstPtr> users = database.users();
+        QCOMPARE(users.count(), 2);
+        QCOMPARE(users[0]->userId(), QLatin1String("a"));
+        QCOMPARE(users[0]->updatedTime(), time1);
+        QCOMPARE(users[0]->userName(), QLatin1String("c"));
+        QCOMPARE(users[0]->count(), 0);
+
+        QCOMPARE(users[1]->userId(), QLatin1String("d"));
+        QCOMPARE(users[1]->updatedTime(), time2);
+        QCOMPARE(users[1]->userName(), QLatin1String("f"));
+        QCOMPARE(users[1]->count(), 0);
+
+/*        OneDriveImageCacheModel model;
```

Comment:
Accidently, actually :) It passes fine, was doing some debugging during development and forgot it like that. Will fix.